lib/services/services.c

```diff
@@ -305,7 +335,28 @@ resources_action_create(const char *name, const char *standard,
     }
     services_action_free(op);
 
-    return NULL;
+    if (op) {
+        return op;
+    } else {
+        return NULL;
+    }
```

Comment:
We don't want to free params and op anymore if we're returning op :) which I guess means we don't even need return_error anymore, we can just return op on error